site/ja/tfx/index.md

```diff
@@ -0,0 +1,585 @@
+    min_count: 1
+  }
+}
+feature {
+  name: "capital-gain"
+  value_count {
+    min: 1
+    max: 1
+  }
+  type: FLOAT
+  presence {
+    min_fraction: 1
+    min_count: 1
+  }
+}
+...
+```
+
+スキーマを利用しているコンポーネントを次に示します :
+
+- TensorFlow Data Validation
+- TensorFlow Transform
+
+典型的な TFX のパイプラインでは TensorFlow Data Validation がスキーマを生成し、
+ほかのコンポーネントはそれを利用します。
+
+Note: 自動生成されたスキーマは「ベストエフォート」のものであり、単にデータの基本的な特徴の推測を試みるものです。
+生成されたスキーマについては、開発者が確認し、必要な場合は修正を加えることが期待されています。
+
+## TFX を使った開発
+
+TFX は機械学習のプロジェクトやリサーチ、実験、ローカルのマシン上での開発、デプロイを行うまでのすべてのフェーズにおいて、強力なプラットフォームを提供します。
+コードの重複を避け、[学習時とサービス提供時の歪み](#training-serving-skew-detection) の可能性を排除するために、
+学習時と学習済みモデルのデプロイ時の両方で TFX のパイプラインを実装し、[TensorFlow Transform](tft.md) ライブラリを
+を活用するために [Transform](transform.md) コンポーネントを学習時と推論時の両方で利用することを強く推奨します。
+このようにすることで、前処理や解析を行う同一のコードを一貫して利用することができ、
+学習に利用するデータとプロダクション環境で学習済みモデルに与えられるデータの間で差異が生じることを避けられます。
+また、コードの記述も一度で済みます。
+
+### データ探索、可視化、クリーニング
+
+![Data Exploration, Visualization, and Cleaning](wrangling.svg)
+
+TFX パイプラインは典型的には [ExampleGen](examplegen.md) コンポーネントから始まります。
+ExampleGen コンポーネントは入力されたデータを受け付け、 tf.Examples の形式に整形します。
+これはデータセットが学習用と評価用に分割されたあとで実行されることもよくあるため、学習用と評価用に、
+2 つの ExampleGen コンポーネントのコピーが存在する場合もあります。
+また、一般的には次に [StatisticsGen](statsgen.md) コンポーネントと [SchemaGen](schemagen.md)
+コンポーネントが続きます。
+これらのコンポーネントはデータを確認し、データのスキーマと統計量を推定します。
+スキーマと統計量は [ExampleValidator](exampleval.md) コンポーネントに入力されます。
+このコンポーネントはデータの中に異常値や欠損値、誤ったデータ型が存在しないか検査します。
+これらのコンポーネントは [TensorFlow Data Validation](tfdv.md) の能力を活用しています。
+
+[TensorFlow Data Validation (TFDV)](tfdv.md) はデータセットの探索、可視化、クリーニングを
+行う際に役に立つツールです。
+TFDV はデータを確認してデータの型、カテゴリ、値域を推定し、その後、自動的に異常値や欠損値を特定するのを手助けします。
+パイプラインのコンポーネントの処理が完了したあと、 [MLMD](mlmd.md) からメタデータを読み込み、
+データを分析するために TFDV の可視化ツールを Jupyter ノートブック上で利用できます。
+最初にモデルがデプロイされたあとに、TFDV をデプロイされたモデルへの推論リクエストに含まれる
+新たなデータを監視し、異常値やドリフトの検出を行うために利用できます。
+これは時系列データでトレンドや季節性があり、時間の経過にしたがって変化するものに対してはとくに有効で、
+データの問題やモデルを新しいデータで再学習させる必要があるときに、通知を行う手助けになります。
+
+### データの可視化
+
+パイプラインのうち、TFDV を利用するコンポーネント (典型的には StatisticsGen, SchemaGen, ExampleValidator) では、データを処理した結果を Jupyter notebook で可視化できます。
+データがモデルやアプリケーションに最適化されるまで何度も実行して結果を比較し、調整を行うことも可能です。
+
+これらのコンポーネントの出力結果を [**ML Metadata (MLMD)**](mlmd.md) にクエリを発行して保存したあと notebook で可視化するために、 TFDV の可視化をサポートする API を利用できます。
+これには [tfdv.load_statistics()](`tfdv.load_statistics`) や [tfdv.visualize_statistics()](`tfdv.visualize_statistics`) が含まれます。
+これらの可視化を利用することで、データセットの特徴についてよりよい理解を得ることや、もし必要な場合は要求にそって修正することもできます。
+
+### モデルの開発と学習
+
+![Feature Engineering](feature_eng.svg)
+
+典型的な TFX パイプラインは [Transform](transform.md) コンポーネントを含みます。
+これは [TensorFlow Transform (TFT)](tft.md) ライブラリを活用して特徴量エンジニアリングを行います。
+Transform コンポーネントは SchemaGen コンポーネントの作成したスキーマを入力とし、[data transformations](https://www.tensorflow.org/tfx/transform/api_docs/python/tft) を適用してモデルの学習に利用する特徴量の作成、組み合わせ、変換を行います。
+欠損値の除去や型の変換についても、それらが推論時のリクエストにも含まれる可能性のある場合、 Transform コンポーネントで実行すべきです。
+TensorFlow のコードを TFX で学習させるよう設計するときには[いくつか重要な検討事項があります。](train.md)
```

Comment:
[weak suggestion]
`[いくつか重要な検討事項があります。](train.md)` => `[いくつか重要な検討事項があります](train.md)。`

Reply:
Thank you for your suggestion, however, I do not modify it since other docs (ex. https://www.tensorflow.org/js/tutorials/transfer/audio_recognizer ) uses `[xxx。]` expressions.

Reply:
I see. I thought I found the style `[xxx] 。` in another place myself. However, I cannot find it now.   
It is ok with me to leave this.